DOCSP-40119 Add docs release notes for v20240605 (#774)
mongodb/docs-app-services: source/release-notes/backend.txt

2024 Releases
-------------

.. _backend_20240605:

5 June 2024 Release
~~~~~~~~~~~~~~~~~~~

- Support for nested collections in ``Mixed`` properties is now available. To learn more, refer to :ref:`<collections-in-mixed>`. 

> - Improvements to the Function Editor for all Apps, including the following changes:

Reviewer: nice job on revising this. it reads much smoother 😎

  - A new :guilabel:`Format Code` button to format function code.

  - A new :guilabel:`Clear & Run` button that replace the :guilabel:`Clear Result` button.
  
  - Autocomplete for common functionalities and App Services context.

- App error logs now include GraphQL failures.

> - Users can no longer disable client max offline time. Users can continue to increase or decrease

Reviewer: just confirming again whether we have a corresponding DOCSP ticket for the doc updates on our end

Author: https://jira.mongodb.org/browse/DOCSP-40390

Reviewer: thanks!!

  the number of days for client max offline time where necessary.

> - Device Sync apps with an indexed queryable field now support schemas that do not contain the indexed queryable field. 

Reviewer: sorry if a stupid question, but is this supposed to say something like this:
```suggestion
- Device Sync now supports schemas that do not contain the indexed queryable field. 
```
otherwise, I am pretty confused as to what this feature is actually doing 🙃

Author: Previously, when an app had an IQF, all schemas for the app had to have the IQF as a property. Will add a sentence about the previous behavior to explain.

2024 Releases
-------------

.. _backend_20240605:

5 June 2024 Release
~~~~~~~~~~~~~~~~~~~

> - Atlas Events

Reviewer: 🤔 what are Atlas Events? Should there be additional information here for context?

Author: n/a

>   - Apps with the Atlas Events feature flag will start to react to changes made in their linked Atlas clusters.

Author: Checked with the engineer, deleting Atlas Events, as it is not readily available for public.

  - On cluster deletion, Apps will delete datasources and all resources linked to that datasource for datasources connected to the deleted cluster.

  - When the cluster is updated (instance size change), the next subsequent usage of a Mongo Client will have small 

>     cold start time caused by refreshing client connection details.

Author: n/a

> - Improvements to Function Editing Experience: All Apps will now be able to use a new function editor with autocomplete for common functionalities and App Services context. This includes:
> 
>   - A new button to allow formatting your functions' code
> 
>   - A new clear and run button to replace the Clear result one
>   
>   - Context autocomplete

Reviewer: ```suggestion
- Improvements to Function editor. All Apps can now use the new function editor with autocomplete for common functionalities and App Services context. This includes:

  - A new :guilabel:`BUTTON NAME?` button to format function code.

  - A new :guilabel:`Clear and Run` button that replaces the :guilabel:`Clear Result` button.
  
  - Context autocomplete.
```
is the "Context autocomplete" bullet kind of redundant with the `All Apps can now use the new function editor with autocomplete for common functionalities and App Services context.` sentence?

Author: Thanks, fixed!

> - Collections in mixed data properties is now availbale. See :ref:`<collections-in-mixed>` for more information. 

Reviewer: can we move this to the top of the release notes?
```suggestion
- Support for nested collections in ``Mixed`` properties is now available. To learn more, refer to :ref:`<collections-in-mixed>`. 
```

- Error logs for GraphQL failures now appear in an app's Logs page.

- Users can no longer disable client max offline time. Users can continue to increase or decrease

>   the number of days for client max offline time where neccessary.

Reviewer: Is there a ticket to also update this in our docs (e.g. [here](https://www.mongodb.com/docs/atlas/app-services/sync/go-to-production/optimize-sync-atlas-usage/#client-maximum-offline-time) or [here](https://www.mongodb.com/docs/atlas/app-services/sync/go-to-production/optimize-sync-atlas-usage/#client-maximum-offline-time))?

Author: Thanks, will make a ticket for this as well! I haven't seen one.

> - Querying is now supported for tables that don't contain the Indexed Queryable Field (IQF).

Reviewer: ```suggestion
- Querying is now supported for tables that don't contain an indexed queryable field.
```
Is this only applicable to Device Sync tables? Does this have any additional doc impacts vis-a-vis querying? (Also, I don't see the acronym "IQF" used anywhere in our docs.)

Author: Yes only applicable for Device Sync apps. The engineers seem to refer to it as IQF, interesting that we don't! I'll remove.
I'll look into docs impact, I believe there might be on the sync settings page.
